scripts/check_frontend_coverage_test.py

```diff
@@ -0,0 +1,335 @@
+    def test_get_stanzas_from_lcov_file_file_name_exception(self):
+        self.lcov_items_list = (
+            'SF:\n'
+            'LF:10\n'
+            'LH:5\n'
+            'end_of_record\n'
+        )
+        with self.open_file_swap:
+            with self.assertRaises(Exception) as error:
+                check_frontend_coverage.get_stanzas_from_lcov_file()
+            self.assertEqual(error.exception.message,
+                'The test path is empty or null. '
+                'It\'s not possible to diff the test coverage correctly.')
+            
+    def test_get_stanzas_from_lcov_file_total_lines_exception(self):
+        self.lcov_items_list = (
+            'SF:/opensource/oppia/file.ts\n'
+            'LF:\n'
+            'LH:5\n'
+            'end_of_record\n'
+        )
+        with self.open_file_swap:
+            with self.assertRaises(Exception) as error:
+                check_frontend_coverage.get_stanzas_from_lcov_file()
+            self.assertEqual(error.exception.message,
+                'It wasn\'t possible to get the total lines of file.ts file.'
+                'It\'s not possible to diff the test coverage correctly.')
+
+    def test_get_stanzas_from_lcov_file_covered_lines_exception(self):
+        self.lcov_items_list = (
+            'SF:/opensource/oppia/file.ts\n'
+            'LF:10\n'
+            'LH:\n'
+            'end_of_record\n'
+        )
+        with self.open_file_swap:
+            with self.assertRaises(Exception) as error:
+                check_frontend_coverage.get_stanzas_from_lcov_file()
+            self.assertEqual(error.exception.message,
+                'It wasn\'t possible to get the covered lines of file.ts file.'
+                'It\'s not possible to diff the test coverage correctly.')
+    
+    def test_check_coverage_changes(self):
+        self.lcov_items_list = (
+            'SF:/opensource/oppia/file.ts\n'
+            'LF:10\n'
+            'LH:10\n'
+            'end_of_record\n'
+            'SF:/opensource/oppia/file2.ts\n'
+            'LF:10\n'
+            'LH:10\n'
+            'end_of_record\n'
+        )
+        fully_covered_tests_swap = self.swap(check_frontend_coverage,
+            'fully_covered_filenames', [
+                'file.ts',
+                'file2.ts'
+            ]
+        )
+        with self.exists_swap, self.open_file_swap, self.print_swap:
+            with fully_covered_tests_swap:
+                check_frontend_coverage.check_coverage_changes()
+                self.assertEqual(
+                    self.check_function_calls_counter['mock_print'], 3)
+
+    def test_check_coverage_changes_error(self):
+        def mock_exists(unused_path):
+            return False
+        exists_swap = self.swap(os.path, 'exists', mock_exists)
+        with exists_swap:
+            with self.assertRaises(Exception) as error:
+                check_frontend_coverage.check_coverage_changes()
+            self.assertTrue(error.exception.message)
```

Comment:
Check the text of the message. You can do self.assertRaisesRegexp.

Reply:
Done.